tests/nmap_wrapper_test.py

```diff
@@ -0,0 +1,95 @@
+"""Nmap wrapper unit tests"""
+import pytest
+
+import agent.nmap_agent
+from agent import nmap_options
+from agent import nmap_wrapper
+
+
+@pytest.mark.parametrize(
+    "nmap_parametrized_agent",
+    [["fast_mode.yaml"]],
+    indirect=True,
+)
+def testNmapWrapper_whenFastMode_returnCommand(
+    nmap_parametrized_agent: agent.nmap_agent.NmapAgent,
+) -> None:
+    args = nmap_parametrized_agent.args
+    options = nmap_options.NmapOptions(
+        dns_resolution=False,
+        ports=args.get("ports"),
+        top_ports=args.get("top_ports"),
+        fast_mode=args.get("fast_mode", False),
+        no_ping=args.get("no_ping", False),
+        timing_template=nmap_options.TimingTemplate[args["timing_template"]],
+        scripts=args.get("scripts"),
+        script_default=args.get("script_default", False),
+        version_detection=args.get("version_info", False),
+    )
+    client = nmap_wrapper.NmapWrapper(options)
+
+    command = client.construct_command_host("127.0.0.1", 24)
+
+    assert "-F" in command
+    assert "--top-ports" not in command
+    assert "-p" not in command
+
+
+@pytest.mark.parametrize(
+    "nmap_parametrized_agent",
+    [["top_ports.yaml"]],
```

Comment:
This is not how we test the config, because those files are not maintainable, you should pass the config instead.